### ISSUE TITLE
Cannot expand variable in comment

### ISSUE THREAD

**Issue by gromgull** [pugjs/pug#1311]
Hi, 

I'd like to have a variable expanded inside a comment for inserting a cloudfront server in a useminPrepare block: 

```
// build:js #{cloudfront}js/app.min.js
```

Just outputs the #{} verbatim: 

``` html
    <!-- build:js(adminapp.src) #{cloudfront}js/testapp.min.js-->
```

```
  //
      = " build:js "+cloudfront+"js/app.min.js"
```

nearly works, but the closing comments ends up on a new line, which confuses useminPrepare:

``` html
<!-- build:js(adminapp.src) http://t/js/testapp.min.js
    -->
```

Is there any way around this?

**Comment by monolithed** [comment]
The comments are not displayed in the browser, but it still full-fledged tags!

**Comment by ForbesLindesay** [comment]
Because generated comments are currently second class citizens.  Comments are designed for commenting rather than anything else.

**Comment by monolithed** [comment]
Template engines are designed to generate HTML code.

Even the JavaScript API allows us to fully manipulate comments:

```js
function getCommentNodes(node) {
    var treeWalker = document.createTreeWalker(node,
        NodeFilter.SHOW_COMMENT, null, false),

        comments = [];

  while (treeWalker.nextNode()) {
        comments.push(treeWalker.currentNode);
  }

  return comments;
}

// Create a comment
var comment = document.createComment('foo');
document.body.appendChild(comment);

// Get all comments
var comments = getCommentNodes(document.body);

// Iterate over comments
comments.forEach(function (comment) {
  console.log(comment.data); // foo
});
```

I know a lot of template engines that support my point of view:

#### [HAML](http://haml.info/):
```haml
/
  = variable

/
  #{variable}
```

#### [EJS](http://www.embeddedjs.com/)
```html
 <!-- <%= variable %> -->
```

#### [Dust](http://akdubya.github.io/dustjs/)
````html
<--! {variable} -->
```

#### [Mustache](http://mustache.github.io/), [Handlebars](handlebarsjs.com), [Hogan](http://twitter.github.io/hogan.js/)
````html
<!-- {{variable}} -->
```

#### [Fest](https://github.com/mailru/fest)
````html
<fest:comment>
    <fest:value>variable </fest:value>
</fest:comment>
```

I can not even remember any template engines that does not support interpolation comments. Only Jade?

**Comment by dbeff** [comment]
+1 @monolithed

**Comment by dopatraman** [comment]
why was this closed without resolution

**Comment by bickle66** [comment]
I have to agree. I use gulp-useref. For cache busting I would love to be able to:

```
// build:js /js/app.min.js?v=#{version}
script(src="/js/app.js?v=" + version)
// endbuild
```

**Comment by ForbesLindesay** [comment]
Because you can just do:


```html
<!--build:js /js/app.min.js?v=#{version}-->
<!--endbuild-->
```

If you really need that and it's perfectly valid jade.  It's also much better to do it this way if you're using this for builds since a lot of build tools that use comments tend to be really sensitive about precise spacing etc. and this gives you total control.

**Comment by Ulrikop** [comment]
I searched for the variable expanding at comments for the same reason and I found that issue.

I tried the solution with writting the comment directly in html:

    
    

The problem is, that I have something above the block:

    meta(charset="utf-8")

    
    

After compiling pug the result is:

    <meta charset="utf-8">
    

The problem is that usemin removes all before build:css and I lose the meta tag (I use already the conig `pretty = true`)

I can add `= "\n"` but that is ugly. What is the prefered way?
